File: Leetcode-Problems/Kth Distinct String in an Array.cpp
class Solution
{
public:
    string kthDistinct(vector<string> &arr, int k)
    {
        unordered_map<string, int> mp;
        for (string s : arr)
        {
            mp[s]++;
        }
        int count = 0;
        for (string s : arr)
        {
            if (mp[s] == 1)
            {
                count++;
                if (count == k)
                {
                    return s;
                }
            }
        }
        return "";
    }
};

File: Leetcode-Problems/Kth Distinct String in an Array.java
class Solution {
    public String kthDistinct(String[] arr, int k) {
        HashMap<String, Integer> mp = new HashMap<>();
        for (String s : arr) {
            mp.put(s, mp.getOrDefault(s, 0) + 1);
        }
        int count = 0;
        for (int i = 0; i < arr.length; i++) {
            if (mp.get(arr[i]) == 1) {
                count++;
            }
            if (count == k) {
                return arr[i];
            }
        }
        return "";
    }
}